core/src/main/java/bio/terra/pearl/core/service/notification/NotificationService.java

```diff
@@ -56,6 +56,19 @@ public List<Notification> findAllByConfigId(UUID configId, boolean withEnrollees
             attachEnrollees(notifications);
         }
 
+        // assumption: not all notifications have sendgrid events, so we have to manually find the notification
+        //             by id. even though this the number of notifications likely on the order of thousands,
+        //             it shouldn't be too slow since it is only one DB call.
+        sendgridEventDao.findByTriggerId(configId)
+                .forEach(sendgridEvent -> {
+                    Notification notification = notifications
+                            .stream()
+                            .filter(n -> n.getId().equals(sendgridEvent.getNotificationId()))
+                            .findFirst()
+                            .orElseThrow(() -> new RuntimeException("Sendgrid event without notification"));
+                    notification.setEventDetails(sendgridEvent);
+                });
```

Comment:
```suggestion
       List<SendgridEvent> events = sendgridEventDao.findAllByPropertyCollection("notification_id", notifications.stream.map(Notification::getId).toList);
     // now assign them to the right event   
```

Reply:
Hmm. I am not sure how I feel about creating a SQL query with 1000+ clauses in the where, I assumed that would be undesirable. I'm happy to refactor it to this, though.